f/KAS-3431 Vrijgave: automanual

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-3431)

#### Zusterbranches
[app-kaleidos](https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/297)
[yggdrasil](https://github.com/kanselarij-vlaanderen/yggdrasil/pull/37)
[document-release](https://github.com/kanselarij-vlaanderen/document-release-service/pull/2)
[themis-export](https://github.com/kanselarij-vlaanderen/themis-export-service/pull/5)
[newsletter-service](https://github.com/kanselarij-vlaanderen/newsletter-service/pull/45)
[agenda-approve](https://github.com/kanselarij-vlaanderen/agenda-approve-service/pull/27)

#### Inhoud
- models/-wijzigingen: meeting, en 3 types x-publication-activity
- routes + components:
  - /agendas
      - EditMeetingModal: "Publicatiedatum"
  - /vergadering
    - AgendaActionsComponent:
       - nieuwe actie "Vrijgeven en publiceren" (trigger AgendaPublicationPlanModal)
       - releaseDocuments/releasedDecisions-DateTimes -> XPublicationActivity.plannedPublicationTime
       - rekening houden met mogelijkheid niet-gestarte PublicationActivities (vb. `latestPublicationActivity`
       - renames van release naar publish
      - AgendaPublicationsPlanModalComponent
  - /kort-bestek
     - verwijderen EditNewsletterComponent
     - datums tonen
- services/
    - createNewsItemForMeeting: aanpassen aan nieuwe datum-attributen.
- utils/
    - DateUtils.getNextWorkday
    - AgendaPublicationUtils: enkele gedeelde functies

De wijzigingen in het data-model leiden tot nieuwe situaties, niet beschreven in het ticket. Ze zijn al wel besproken met Tom.
#### Opgelost
- `InternalDecisionPublicationActivity`
   - Geen functionele aanpassing.
- `InternalDocumentPublicationActivity`
   - Een interne document-publicatie kan een verschillende `unconfirmedPublicationTime` en `plannedPublicationTime` hebben.
       oplossing: in de `EditMeetingModal` staat de input `unconfirmedPublicationTime` enkel bij aanmaak van de meeting
   - Een interne document-publicatie kan nu naar een datum in de toekomst gezet worden.
       - Yggdrasil start niet automatisch bij verstrijken datum.
         oplossing: document-release-service triggert Yggdrasil door `prov:startedAtTime` te INSERTen
- `ThemisPublicationActivity`

#### Onopgelost
- `InternalDecisionPublicationActivity`
   - Geen functionele aanpassing.
- `InternalDocumentPublicationActivity`
   - Een interne document-publicatie kan nog geen geplande publicatietijd hebben
       onopgelost: datum wordt verwacht in de newsletter-service
   - Een interne document-publicatie-datum kan gewijzigd worden
       onopgelost: datum wordt gebruikt in de newsletter-service
   - Een interne document-publicatie kan nu naar een datum in de toekomst gezet worden.
       onopgelost: mag Themis-publicatie dan gestart worden in `/kort-bestek`? indien ja, ook met scope `documents`?
       onopgelost: mag de newsletter dan verstuurd worden in `/kort-bestek`?
- `ThemisPublicationActivity`
   - Functionele aanpassing: deze datum kan in de toekomst liggen.
       onopgelost: mag Themis-publicatie dan gestart worden in `/kort-bestek`? indien ja, ook met scope `documents`?
       onopgelost: mag de newsletter dan verstuurd worden in `/kort-bestek`?

Commit:
KAS-3431 (auk) accept class attribute | Auk::FormHelpText
Release decisions: Automanual

Commit:
KAS-3431 Release decisions: Automanual: model changes
Release decisions: Automanual

Commit:
KAS-3431 Release decisions: Automanual: model changes
Release decisions: Automanual

Commit:
KAS-3431 remove unused properties of MeetingModel
Release decisions: Automanual

Commit:
KAS-3431 add "Publicatiedatum" to EditMeetingModal
Release decisions: Automanual

Commit:
KAS-3431 update AgendaActions for decision release
Release decisions: Automanual

Commit:
KAS-3431 update AgendaActions for planned publication
Release decisions: Automanual

Commit:
KAS-3431 use PUBLICATION_PROCESSING_WINDOW env var
Release decisions: Automanual

Commit:
KAS-3431 remove edit modal newsletter/final
Release decisions: Automanual

Commit:
KAS-3431 (name) prepend model names with internal- + fix constant THEMIS_PUBLICATION_SCOPE_INITIAL
Release decisions: Automanual

Commit:
KAS-3431 (name) fix: prepend model names with internal-
Release decisions: Automanual

Commit:
KAS-3431 fix: cannot read property of undefined / ::AgendaPublicationPlanModal
Release decisions: Automanual

Commit:
KAS-3431 planned themis publication is first with scope [newsitems, documents] / ::AgendaActions
Release decisions: Automanual

Commit:
KAS-3431 DDAU savePlanPublication / ::AgendaActions
Release decisions: Automanual

Commit:
KAS-3431 fix: loadLatestPublicationActivity -> loadPublicationActivities / ::AgendaActions
Release decisions: Automanual

Commit:
KAS-3431 fix: scope `undefined` / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 update publication dates to activity based data model / NewsletterMeetingComponent
Release decisions: Automanual

Commit:
KAS-3431 updates / EditMeetingModalComponent
- @min check
- rename
- cleanup
Release decisions: Automanual

Commit:
KAS-3431 (tra) consistent en vs. &
Release decisions: Automanual

Commit:
KAS-3431 simplify canPlanPublication check / AgendaActionsComponent
Release decisions: Automanual

Commit:
KAS-3431 fix: canPublishThemis check / AgendaActionsComponent
Release decisions: Automanual

Commit:
KAS-3431 com: avoid term legacy* / AgendaActionsComponent
* probably confusing
Release decisions: Automanual

Commit:
KAS-3431 remove margin from publication processing window / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 cst: / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 update createNewsItemForMeeting to new data model / NewsletterService
Release decisions: Automanual

Commit:
KAS-3431 make plannedPublicationDate required / EditMeetingModal
Release decisions: Automanual

Commit:
KAS-3431 und: remove plannedPublicationDate required asterisk / EditMeetingModal
Release decisions: Automanual

Commit:
KAS-3431 include XPublicationActivities / NewsletterPrintRoute
Release decisions: Automanual

Commit:
KAS-3431 filter on finished activities / NewsletterHeaderOverviewComponent
Release decisions: Automanual

Commit:
KAS-3431 nam: isInternalDocumentPublished / AgendaActionsComponent
Release decisions: Automanual

Commit:
KAS-3431 add getIsPublished / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime - step 1: models
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime - step 2:
- replace plannedStart -> unconfirmedPublicationTime
- replace startDate -> plannedPublicationTime
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime - step 3: rename EditMeetingModal.plannedPublicationDate -> unconfirmedPublicationTime
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime - fix: use internalDecisionPublicationActivity.startDate / AgendaActionsComponent
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 repurpose XPublicationActivity.xStartTime - update is published check for internalDecisionPublicationActivity / AgendaActionsComponent
Release decisions: Automanual

Commit:
Merge branch 'development' into f/KAS-3431-release-decisions--automanual--1

Commit:
KAS-3431 tra: cleanup
Release decisions: Automanual

Commit:
KAS-3431 show publication times / NewsletterMeetingComponent
Release decisions: Automanual

Commit:
KAS-3431 handle legacy data / NewsletterService.createNewsItemForMeeting
Release decisions: Automanual

Commit:
KAS-3431 add TODO? / AgendaPublicationUtils
Release decisions: Automanual

Commit:
KAS-3431 fix: plannedStart -> plannedPublicationTime / AgendaActionsComponent
Release decisions: Automanual

Commit:
KAS-3431 default empty scope / ThemisPublicationActivityModel
Release decisions: Automanual

Commit:
KAS-3431 AgendaPublicationUtils + NewsletterMeetingComponent
Release decisions: Automanual

Commit:
KAS-3431 cle / NewsletterMeetingComponent
Release decisions: Automanual

tdn commented on app/components/agenda/agenda-header/agenda-actions.js:
```diff
@@ -51,32 +56,56 @@ export default class AgendaAgendaHeaderAgendaActions extends Component {
+    if (this.loadPublicationActivities.isRunning) return false;
+
+    let can = this.currentSession.isEditor;
+    can &&= this.args.meeting.isFinal;
```
to clarify, this does:
`can = can && this.args.meeting.isFinal`

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/frontend-kaleidos into f/KAS-3431-vrijgave--automanual

ValenberghsSven commented:
Ik ga dit al even noteren om niet te vergeten + eventueel te discussieren.
In de backend kant zijn `releasedDecisions` en `releasedDocuments` attributen van meeting verwijderd.
Maar in frontend zijn ze nog gebruikt (ook actief in nieuwe code voor dit ticket, veroorzaakte errors bij saven)
Ik vermoed dat we daar moeten overschakelen naar die modellen hun `plannedPublicationTime`

ValenberghsSven commented on app/components/newsletter/newsletter-meeting.hbs:
```diff
@@ -4,51 +4,25 @@
-        (await this.meeting.newsletter.publicationDate)
-        "DD MMMM YYYY - HH:ss"
-      }}
+      {{this.formatPublicationTime this.internalDocumentPublicationTime}}
```
Dit is niet correct volgens mij.
De `newsletter.publicationDate` is normaal de datum van de vergadering om 14:00.
Dit kon vroeger enkel gewijzigd worden in de KB views (klad/definitief)
deze waarde != de interne vrijgave van documenten.

ValenberghsSven committed:
KAS-3431 combined commit WIP

add release status, remove unconfirmed partially, edit some of the models to match the resources, change newsletter header action to work with status when creating themis-publcation

ValenberghsSven committed:
KAS-3431 change create and edit of meeting, make util for document released check, add todos

erikap committed:
Merge branch 'development' into f/KAS-3431-vrijgave--automanual

erikap committed:
Update publication activity models

- Refactor release-status to generic concept
- Rename plannedPublicationTime attribute to plannedDate

erikap committed:
Refactor setting planned date in edit-meeting-modal

erikap committed:
Set default access-level to MR when uploading file on confidential
subcase

Logic was currently only implemented when uploading a document from
the agendaitem side, not from the subcase side.

erikap committed:
Refactor check whether documents are already released for overheid profile

erikap committed:
Embed newsletter-header component in newsletters template

Simple component without logic that is only used on 1 page.

erikap committed:
Refactor newsletter print header component

erikap committed:
Remove unused agenda-publication utils

erikap committed:
Rename document release to document publication

erikap committed:
Remove TODO about concurrency on publication-activities

Publication activity models are always reloaded before being edited to
ensure user is working on fresh data

erikap committed:
Disable publication planning when locked in the release window

Planned date can only be changed uptil 30 minutes before the planned date

erikap committed:
Use date helpers from date-fns

erikap committed:
Update planned doc pub date when changing start date of meeting

erikap committed:
Handle null values in pub activities in agenda-actions component

erikap committed:
Define roles to manage decision/document/themis publications

erikap committed:
Fix issues to update dates in document publication planning modal

erikap committed:
Ensure fresh data when opening document publication planning modal

erikap committed:
Check on pub release window must be executed on the persisted planned date

erikap committed:
Undo setting env var in Dockerfile

erikap committed:
Fix fetching latest pub activity in newsletter header

erikap commented:
Triggered a custom Jenkins build http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/260/

erikap committed:
Fix setting planned doc publication date for PVV meeting

erikap committed:
Fix interception of decision/document release requests in Cypress

erikap committed:
Temporarily disable document propagation tests

erikap committed:
Intercept data loading requests when editing meeting in Cypress tests

erikap committed:
Fix typo in intercept request URL in cypress tests
